QRCoder/QRCodeGenerator.cs

```diff
@@ -139,20 +153,32 @@ public static QRCodeData GenerateQrCode(string plainText, ECCLevel eccLevel, boo
                 {
                     var maxSizeByte = capacityTable[version - 1].Details.First(x => x.ErrorCorrectionLevel == eccLevel).CapacityDict[encoding];
                     throw new QRCoder.Exceptions.DataTooLongException(eccLevel.ToString(), encoding.ToString(), version, maxSizeByte);
-                }                    
+                }
             }
 
-            string modeIndicator = String.Empty;
+            var completeBitArrayLength =
+                (eciMode != EciMode.Default ? 16 : 4) +  // Mode indicator
+                GetCountIndicatorLength(version, encoding) + // Count indicator
+                codedText.Length;                            // Data
+
+            var completeBitArray = new BitArray(completeBitArrayLength);
+            // write mode indicator
+            var completeBitArrayIndex = 0;
             if (eciMode != EciMode.Default)
             {
-                modeIndicator = DecToBin((int)EncodingMode.ECI, 4);
-                modeIndicator += DecToBin((int)eciMode, 8);
+                DecToBin((int)EncodingMode.ECI, 4, completeBitArray, ref completeBitArrayIndex);
+                DecToBin((int)eciMode, 8, completeBitArray, ref completeBitArrayIndex);
+            }
+            DecToBin((int)encoding, 4, completeBitArray, ref completeBitArrayIndex);
+            // write count indicator
+            DecToBin(dataInputLength, GetCountIndicatorLength(version, encoding), completeBitArray, ref completeBitArrayIndex);
```

Comment:
Second call to GetCountIndicatorLength. (First call in line 161.) Shall we store the result of the first call and re-use the variable with the result here?

Reply:
Good idea.  Done.